core/eolearn/tests/test_core_tasks.py

```diff
@@ -229,49 +229,37 @@ def test_duplicate_feature_fails(patch: EOPatch) -> None:
-
-    with pytest.raises(ValueError):
-        # Expected a ValueError when trying to initialize a feature with a wrong shape dimensions.
-        patch = InitializeFeatureTask((FeatureType.MASK_TIMELESS, "wrong"), shape=shape, init_value=init_val)(patch)
-
-    init_val = 123
-    shape = (10, 10, 3)
-    compare_data = np.ones(shape) * init_val
-
-    patch = InitializeFeatureTask((FeatureType.MASK_TIMELESS, "test"), shape=shape, init_value=init_val)(patch)
-    assert patch.mask_timeless["test"].shape == shape
-    assert np.array_equal(patch.mask_timeless["test"], compare_data)
-
+@pytest.mark.parametrize(
+    "init_val, shape, feature_type, names, case_feature_spec",
+    [
+        (123, (5, 10, 10, 3), FeatureType.MASK, "test", False),
+        (123, (10, 10, 3), FeatureType.MASK_TIMELESS, "test", False),
+        (123, (5, 10, 10, 3), FeatureType.MASK, ("F1", "F2", "F3"), False),
+        (123, (FeatureType.DATA, "bands"), FeatureType.DATA, ("F1", "F2", "F3"), True),
+    ],
+)
+def test_initialize_feature(
+    init_val: float,
+    shape: Union[Tuple[int, ...], FeatureSpec],
+    feature_type: FeatureType,
+    names: Union[str, Tuple[str, ...]],
+    patch: EOPatch,
+    case_feature_spec: bool,
+) -> None:
+    result_shape = patch[shape].shape if case_feature_spec else shape
+    expected_data = init_val * np.ones(result_shape)
+
+    patch = InitializeFeatureTask({feature_type: names}, shape=shape, init_value=init_val)(patch)
+    assert all([np.array_equal(patch[feature_type][key], expected_data) for key in names])
+
+
+def test_initialize_feature_fails(patch: EOPatch) -> None:
     with pytest.raises(ValueError):
         # Expected a ValueError when trying to initialize a feature with a wrong shape dimensions.
-        patch = InitializeFeatureTask((FeatureType.MASK, "wrong"), shape=shape, init_value=init_val)(patch)
-
-    init_val = 123
-    shape = (5, 10, 10, 3)
-    compare_data = np.ones(shape) * init_val
-    new_names = ("F1", "F2", "F3")
-
-    patch = InitializeFeatureTask({FeatureType.MASK: new_names}, shape=shape, init_value=init_val)(patch)
-    assert set(new_names) < set(patch.mask), "Failed to initialize new features from a shape tuple."
-    assert all(patch.mask[key].shape == shape for key in new_names)
-    assert all(np.array_equal(patch.mask[key], compare_data) for key in new_names)
-
-    patch = InitializeFeatureTask({FeatureType.DATA: new_names}, shape=(FeatureType.DATA, "bands"))(patch)
-    assert set(new_names) < set(patch.data), "Failed to initialize new features from an existing feature."
-    assert all(patch.data[key].shape == patch.data["bands"].shape for key in new_names)
+        InitializeFeatureTask((FeatureType.MASK_TIMELESS, "wrong"), (5, 10, 10, 3), 123)(patch)
 
     with pytest.raises(ValueError):
-        InitializeFeatureTask({FeatureType.DATA: new_names}, 1234)
+        InitializeFeatureTask({FeatureType.DATA: ("F1", "F2", "F3")}, 1234), "Shape argument should be a shape tuple."
```

Comment:
i dont understand this test

Reply:
removed (because of typing) it was old test that test raising ValueError in case of incorrect input